### PR TITLE
track user consent and update cookie banner ux

### PR DESCRIPTION
**Pull request by kevsers**
#### What's this PR do? Any additional context?
Adds the JS to track the user consent
Updates the ux for the cookie-banner

#### Where should the reviewer start?
Start from the cookieBanner.js

#### How can this be manually tested? (brief repro steps)
1- Clear all cookies in your browser
2- Launch the LMS site
3- Verify the cookie-banner exist
4- Click around (to nonlinks) and move mouse around and ensure the cookie banner stays
5- Click on the LearnMore link on banner, ensure the banner stays still
6- Click on a link on the page and ensure the banner is now removed and cookie cretaed

#### What are the relevant TFS items? (list id numbers)
91631 / 96915
#### Definition of done:
- [x] Title of the pull request is clear and informative
- [x] Add pull request hyperlink to relevant TFS items
- [x] For large or complex change: schedule an in-person review session

[//]: # ( todo: Is there appropriate test coverage? )
[//]: # ( todo: Does this PR require a new Selenium test? )
[//]: # ( todo: Is there appropriate logging/monitoring included? )

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging into the development branch then "flatten" or "squash" commits
1. If merging from development into master then don't "flatten" or "squash" commits

#### Reminders AFTER merging
1. Delete the remote branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc)

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 3 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 4 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 5 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

File: comprehensive/lms/static/cookieBanner.js
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: i'd highly recommend registering an event "listener" on this id instead

Comment (sdolenc) [reply comprehensive/lms/static/cookieBanner.js]: oh i see. i misread the code. i first thought the final comparison operator was equality, but it's actually inequality. i understand why it's written this way now, but i'm pretty sure there is another way of doing this. i'll think about it some more and get back to you

**Comment by sdolenc** [comment]
can you share the URL of an instance with this code running? I'd love to play with it as part of the review. let me know if not and i'll checkout the branch, etc

File: comprehensive/lms/static/cookieBanner.js
Comment (sdolenc) [reply comprehensive/lms/static/cookieBanner.js]: okay - i've thought about it some more. i think we'd be better off registering two click event listeners (on hyperlinks and buttons). the implementation function would ensure the element doesn't have "btnprivacy" id before dismissing the banner. this should result in about the same lines of code, but should be much more performant because we won't exhaust the whole DOM tree on each click event AND I think it'll be more readable. We can also remove the listeners when the cookie banner is closed.

**Review by sdolenc — CHANGES_REQUESTED** [review]
good work, but please take a look at my suggestion when you get the chance. thanks :)

File: comprehensive/lms/static/cookieBanner.js
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: please remove "debugger" before merging (and the one below)

File: comprehensive/lms/templates/navigation.html
Comment (sdolenc) [inline comprehensive/lms/templates/navigation.html]: nit: i generally prefer triple quality === , but i'm not blocking on this
https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

File: comprehensive/lms/static/cookieBanner.js
Comment (sdolenc) [reply comprehensive/lms/static/cookieBanner.js]: i'm blocking on this

**Review by sdolenc — CHANGES_REQUESTED** [review]
change looks amazing! please remove the "debugger" keywords when you get the chance and i'll approve. great work 👍 !

File: comprehensive/lms/templates/navigation.html
Comment (sdolenc) [reply comprehensive/lms/templates/navigation.html]: also, do we need to inline js in the html file? i'd rather move this to the js file if possible. in fact, we don't have to register the on click events if this conditional is true. in other words, if (cookie) { closeBanner(); } else { /*register on click events*/ }

the jquery off() will should be a no-opt if on() doesn't happen (but we should verify that) so it should be safe to execute off() everytime closeCookieBanner() is called even if on() wasn't called (as the code is written now)

i think this would be a good improvement.

File: comprehensive/lms/static/cookieBanner.js
Comment (kevsers) [reply comprehensive/lms/static/cookieBanner.js]: removed, artifact from testing...

File: comprehensive/lms/templates/navigation.html
Comment (kevsers) [reply comprehensive/lms/templates/navigation.html]: good info for the triple equality!
the inlining was part of the earlier impl. so moved that logic to the JS file.
And yes, calling off multi times is a no-op.

File: comprehensive/lms/static/cookieBanner.js
Comment (kevsers) [reply comprehensive/lms/static/cookieBanner.js]: and thanks for the amazing review!

**Review by ms-hukoyun — APPROVED** [review]

**Review by satyarapelly — APPROVED** [review]

**Review by sdolenc — APPROVED** [review]

**Pull request merged**
